org.eclipse.lsp4e/src/org/eclipse/lsp4e/ILSWrapper.java

```diff
@@ -0,0 +1,66 @@
+package org.eclipse.lsp4e;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.lsp4j.ServerCapabilities;
+import org.eclipse.lsp4j.services.LanguageServer;
+
+/**
+ * Represents a running language server, with access to the server's capabilities and
+ * methods to dispatch requests/notifications.
+ *
+ */
+public interface ILSWrapper {
```

Comment:
For consistency, I think we should name this `ILanguageServerWrapper`, without abbreviation.

Reply:
Sure, if that's the consensus. I didn't go for the obvious name because this isn't *necessarily* tied to just exposing bits of `LanguageServerWrapper` - this could be a different class altogether that wraps/delegates to the existing code. I toyed with the idea of a `SingleServerExecutor` object, but that's maybe overkill. So long as there's a clear distinction between running requests on a LS and doing harmless things like querying for capabilities, and doing more low-level lifecycle-related stuff like explicitly starting and stopping them, which I want to keep as separate specialist code.

Reply:
maybe `IServerExecutor`? I would be fine with that or other name, but it looks to me that we do not use abbreviations for java classes, which is an usual java convention, so I would just avoid `ILSWrapper`.

Reply:
Renamed to ILanguageServerWrapper